File: 最小路径和.go
func minPathSum(grid [][]int) int {
	n:= len(grid)
	m:=len(grid[0])

	f := make([][]int, n)
	for i := 0; i < len(dp); i++ {
		f[i] = make([]int, columns)
}
	f[0][0] = grid[0][0]

	for j:=1;j<m;j++ {
    f[0][j] = f[0][j-1]+ grid[0][j]
	}

	for i:=1;i<n;i++ {
		f[i][0] = f[i-1][0]+grid[i][0]
	}

	for i:=1;i<n;i++ {
		for j:=1;j<m;j++ {
			f[i][j] = min(f[i-1][j],f[i][j-1])+ grid[i][j]
		}
	}


}

func min(a,b int) int {
	if a<b{
		return a
	}else {
		return b
	}
}

File: 最小路径和.js
// 动态规划
// f[0][j] = f[0][j-1]+grid[j][0]
// f[i][0] = f[i-1][0]+grid[i][0]
// f[i][j] = min(f[i-1][j],f[i][j-1])+grid[i][j]
/**
 * @param {number[][]} grid
 * @return {number}
 */
 var minPathSum = function(grid) {
  var n = grid.length;
  var m = grid[0].length;
  var f = new Array(n).fill(0).map(()=>new Array(m).fill(0));
    f[0][0] = grid[0][0]

  for(let j =1; j<m;j++){
    f[0][j] = f[0][j-1]+ grid[0][j]
  }
  for(let i = 1; i<n;i++) {
    f[i][0] = f[i-1][0]+ grid[i][0]
  }

  for(let i = 1;i<n;i++){
    for(let j = 1;j<m;j++) {
      f[i][j] = Math.min(f[i-1][j],f[i][j-1])+ grid[i][j]
    }
  }
  return f[n-1][m-1]
};